open-ce/utils.py

```diff
@@ -219,6 +219,74 @@ def cuda_driver_installed():
 
         raise OpenCEError(Error.ERROR, "lsmod command unexpectedly failed") from err
 
+def check_cuda_version_match():
+    '''
+    Check whether cudatoolkit matches the system CUDA version
+    Will return "True" if the match, "False" if not (or err on command error)
+    '''
+    # Approach:
+    #   "nvcc --version" will show cudatoolkit version in use
+    #   nvidia-smi output shows CUDA Version on the system
+
+    # Get version of nvcc (representing cudatoolkit)
+    try:
+        nvcc_output = subprocess.getoutput("nvcc --version")
+        print(nvcc_output)
+        nvcc_version = re.search(r"release (\d+\.\d+)", nvcc_output).group(1)
+    except OSError as err:
+        if err.errno == errno.ENOENT:
+            raise OpenCEError(Error.ERROR, "nvcc command not found") from err
+
+        raise OpenCEError(Error.ERROR, "nvcc command unexpectedly failed") from err
+
+    # Get installed version of CUDA
+    try:
+        smi_output = subprocess.check_output("nvidia-smi").decode("utf-8").strip()
+        cuda_version = re.search(r"CUDA Version\: (\d+\.\d+)", smi_output).group(1)
+    except OSError as err:
+        if err.errno == errno.ENOENT:
+            raise OpenCEError(Error.ERROR, "nvidia-smi command not found") from err
+
+        raise OpenCEError(Error.ERROR, "nvidia-smi command unexpectedly failed") from err
+
+    if nvcc_version == cuda_version:
+        return True
+
+    # Versions do not match.  Raise an error
+    raise OpenCEError(Error.ERROR, "nvcc version does not match system's CUDA version")
+
+def check_cuda_home():
+    '''
+    Ensure CUDA_HOME is set with same PATH as nvcc.
+      If unset, then default it to the PATH of nvcc
+      If paths are different, treat as fatal error.
+    '''
+    try:
+        nvcc_path = subprocess.getoutput("/usr/bin/which nvcc")
```

Comment:
This shouldn't be an error. It could be that `nvcc` is installed under `$CUDA_HOME/bin`, but that hasn't been added to the path.

Reply:
Let's not check whether `nvcc` is in the system path.

Reply:
`nvcc` is no longer invoked at all. We use $CUDA_HOME if set, or set specifically to default of `/usr/local/cuda` if not previously set. (So, no failure if path doesn't match `nvcc` in system path).